x/stake/keeper/validator.go

```diff
@@ -195,12 +194,17 @@ func (k Keeper) RemoveValidator(ctx sdk.Context, address sdk.ValAddress) {
 		panic("Cannot call RemoveValidator on bonded or unbonding validators")
 	}
 
+	// if any tokens remain, remove from pool.
+	// this happens if shares are zero but tokens are not.
+	pool := k.GetPool(ctx)
+	pool.LooseTokens = pool.LooseTokens.Sub(validator.Tokens)
+	k.SetPool(ctx, pool)
```

Comment:
We should probably indicate "burn" tokens here - because that's what's happening. 
Also FYI this shouldn't be necessary once the Dec->Int Update is made due to an update to `RemoveDelegatorShares` https://github.com/cosmos/cosmos-sdk/pull/2958/files#diff-ff4682cf67c96b47c013393f099d1d0eR407

Reply:
Updated comment, added TODO.